File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/contracts/PoPEndorsementInfo.java
// VeriBlock PoP Miner
// Copyright 2017-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.miners.pop.contracts;

import com.google.gson.annotations.SerializedName;
import nodecore.miners.pop.common.Utility;

public class PoPEndorsementInfo {
    public PoPEndorsementInfo(final nodecore.api.grpc.PoPEndorsementInfo popEndorsementInfo) {
        minerAddress = Utility.bytesToBase58(popEndorsementInfo.getMinerAddress().toByteArray());
        endorsedVeriBlockBlockHash = Utility.bytesToHex(popEndorsementInfo.getEndorsedVeriblockBlockHash().toByteArray());
        containedInVeriBlockBlockHash = Utility.bytesToHex(popEndorsementInfo.getContainedInVeriblockBlockHash().toByteArray());
        veriBlockTransactionId = Utility.bytesToHex(popEndorsementInfo.getVeriblockTxId().toByteArray());
        bitcoinTransaction = Utility.bytesToHex(popEndorsementInfo.getBitcoinTransaction().toByteArray());
        bitcoinTransactionId = Utility.bytesToHex(popEndorsementInfo.getBitcoinTxId().toByteArray());
        bitcoinBlockHeader = Utility.bytesToHex(popEndorsementInfo.getBitcoinBlockHeader().toByteArray());
        bitcoinBlockHeaderHash = Utility.bytesToHex(popEndorsementInfo.getBitcoinBlockHeaderHash().toByteArray());
        reward = Utility.formatAtomicLongWithDecimal(popEndorsementInfo.getReward());
        finalized = popEndorsementInfo.getFinalized();
        endorsedBlockNumber = popEndorsementInfo.getEndorsedBlockNumber();
    }

    @SerializedName("miner_address")
    public String minerAddress;

    @SerializedName("endorsed_veriblock_block_hash")
    public String endorsedVeriBlockBlockHash;

    @SerializedName("contained_in_veriblock_block_hash")
    public String containedInVeriBlockBlockHash;

    @SerializedName("veriblock_transaction_id")
    public String veriBlockTransactionId;

    @SerializedName("bitcoin_transaction")
    public String bitcoinTransaction;

    @SerializedName("bitcoin_transaction_id")
    public String bitcoinTransactionId;

    @SerializedName("bitcoin_block_header")
    public String bitcoinBlockHeader;

    @SerializedName("bitcoin_block_header_hash")
    public String bitcoinBlockHeaderHash;

    @SerializedName("reward")
    public String reward;

    @SerializedName("finalized")
    public boolean finalized;

    @SerializedName("endorsed_block_number")
    public int endorsedBlockNumber;
}

File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/contracts/PoPEndorsementInfo.kt
// VeriBlock PoP Miner
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.miners.pop.contracts

import com.google.gson.annotations.SerializedName
import nodecore.api.grpc.VeriBlockMessages
import nodecore.miners.pop.common.Utility

class PoPEndorsementInfo(
    popEndorsementInfo: VeriBlockMessages.PoPEndorsementInfo
) {
    @SerializedName("miner_address")
    val minerAddress: String

    @SerializedName("endorsed_veriblock_block_hash")
    val endorsedVeriBlockBlockHash: String

    @SerializedName("contained_in_veriblock_block_hash")
    val containedInVeriBlockBlockHash: String

    @SerializedName("veriblock_transaction_id")
    val veriBlockTransactionId: String

    @SerializedName("bitcoin_transaction")
    val bitcoinTransaction: String

    @SerializedName("bitcoin_transaction_id")
    val bitcoinTransactionId: String

    @SerializedName("bitcoin_block_header")
    val bitcoinBlockHeader: String

    @SerializedName("bitcoin_block_header_hash")
    val bitcoinBlockHeaderHash: String

    @SerializedName("reward")
    val reward: String

    @SerializedName("finalized")
    val finalized: Boolean

    @SerializedName("endorsed_block_number")
    val endorsedBlockNumber: Int

    init {
        minerAddress = Utility.bytesToBase58(popEndorsementInfo.minerAddress.toByteArray())
        endorsedVeriBlockBlockHash = Utility.bytesToHex(popEndorsementInfo.endorsedVeriblockBlockHash.toByteArray())
        containedInVeriBlockBlockHash = Utility.bytesToHex(
            popEndorsementInfo.containedInVeriblockBlockHash.toByteArray()
        )
        veriBlockTransactionId = Utility.bytesToHex(popEndorsementInfo.veriblockTxId.toByteArray())
        bitcoinTransaction = Utility.bytesToHex(popEndorsementInfo.bitcoinTransaction.toByteArray())
        bitcoinTransactionId = Utility.bytesToHex(popEndorsementInfo.bitcoinTxId.toByteArray())
        bitcoinBlockHeader = Utility.bytesToHex(popEndorsementInfo.bitcoinBlockHeader.toByteArray())
        bitcoinBlockHeaderHash = Utility.bytesToHex(popEndorsementInfo.bitcoinBlockHeaderHash.toByteArray())
        reward = Utility.formatAtomicLongWithDecimal(popEndorsementInfo.reward)
        finalized = popEndorsementInfo.finalized
        endorsedBlockNumber = popEndorsementInfo.endorsedBlockNumber
    }
}
